app/static/src/app/components/WodinPlot.vue

```diff
@@ -59,6 +59,21 @@ export default defineComponent({
             type: Object as PropType<Partial<LayoutAxis> | null>,
             required: false,
             default: null
+        },
+        graphIndex: {
+            type: Number,
+            required: false,
+            default: -1
+        },
+        // We could look up graphConfig from graphIndex - however this causes a specific reactivity bug
+        // when the last plot is deleted and the plot tries to observe the old maximum index before it is updated.
+        graphConfig: {
+            type: Object as PropType<GraphConfig | null>,
+            required: true
+        },
```

Comment:
so i think using the index as a unique identifier is the problem here, i think the only piece of information all these components need to be passed down is the `GraphConfig.id` and use `.find`s or `.findIndex`s to locate where we are, this removes the need for `graphIndex` and `graphConfig` to be passed down and any component can just get that from the store, this will deal with the deleted graph hopefully as you can put that logic after the find. Also the store mutation will be alright too as you can just pass in the id instead of the index into the mutation

on this point i think the logic for updating graph settings and seeing whether we are at the fit plot or not can be extracted out into a composable that gets whether we are fit plot or not straight from the store based on what tab is selected, i think it would be cleaner

thats probably a lot of work so see if you feel  up to these changes or not, happy for this to be ignored

Reply:
Yeah, I definitely had issues with just using index, and tbh I thought needing to use `findIndex` felt dirtier than passing in graphIndex as well as config - which just goes to show that cleanliness can be subjective!

I agree on the composable though, good to keep this dubious logic in one place either way - will make a ticket.

Reply:
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-5572/make-a-composable-for-shared-plot-logic